File: bin/filt_codon_stats.r
#!/usr/bin/env Rscript

args = commandArgs(trailingOnly=TRUE)

if(length(args) < 2){
    stop("Usage: filt_codons_stats.r <dada2_stats> <filtered_table>")
}

dada2_stats <- args[1]
coi_filt <- args[2]

old_stats <- read.table(dada2_stats, header = T, check.names = F, row.names = 1)

filt_table <- read.table(coi_filt, header = T, check.names = F, row.names = 1)

filt_counts <- as.data.frame(colSums(filt_table))
names(filt_counts) <- "codon_filtered"

new_stats <- merge(old_stats, filt_counts, by = 'row.names')

write.table(new_stats, file = "codon.filtered.stats.tsv", row.names=FALSE, sep="\t", col.names = TRUE, quote = FALSE, na = '')

File: bin/filt_codon_stats.py
#!/usr/bin/env python3

import pandas as pd
import sys

# argument check
if len(sys.argv) != 2:
    exit("Usage: count_table_max_reads.py <ASVcounts.tsv>")

# read tsv
data = pd.read_csv(sys.argv[1], sep="\t", skiprows=None)  # count table

# drop feature ids
df = data.drop(data.columns[0], axis=1)

# sum each column
sums = df.sum(axis=0)

# add column with sample names at beginning
out = sums.rename_axis("sample").reset_index()

# rename columns
out = out.rename(columns={0: "encoding"})

# write file
out.to_csv("codon.filtered.stats.tsv", sep="\t", index=False)
